src/index.js

```diff
@@ -3,17 +3,23 @@ import ReactDOM from "react-dom";
 import App from "./App";
 import { makeServer } from "./server";
 import { ProductContextProvider } from "./context/products/productContext";
-import MockAPI from "./mockMan";
-
-
+import { BrowserRouter } from "react-router-dom";
+import { CartContextProvider } from "./context/cart/cartContext";
+import { WishlistContextProvider } from "./context/wishlist/wishlistContext";
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
-    <ProductContextProvider>
-      <App />
-    </ProductContextProvider>
+    <CartContextProvider>
+      <WishlistContextProvider>
+        <ProductContextProvider>
+          <BrowserRouter>
+            <App />
+          </BrowserRouter>
+        </ProductContextProvider>
+      </WishlistContextProvider>
+    </CartContextProvider>
```

Comment:
Just a Suggestion: As your contexts are increased to a certain amount now, you can export them to a single Component too and import Only One Context here.